wom/src/main/scala/wdl4s/wom/graph/InstantiatedExpression.scala

```diff
@@ -0,0 +1,44 @@
+package wdl4s.wom.graph
+
+import cats.instances.list._
+import cats.syntax.traverse._
+import cats.syntax.validated._
+import cats.data.Validated.Valid
+import lenthall.validation.ErrorOr.ErrorOr
+import wdl4s.wdl.types.WdlType
+import wdl4s.wom.expression.WomExpression
+import wdl4s.wom.graph.GraphNodePort.{ConnectedInputPort, InputPort, OutputPort}
+
+class InstantiatedExpression private(val expression: WomExpression, val womReturnType: WdlType, val inputMapping: Map[String, InputPort]) {
+  val inputPorts = inputMapping.values.toSet
+}
+
+object InstantiatedExpression {
+
+  private[graph] def createExpressionNodeLinkedWithInputs[A <: GraphNode](applier: (String, InstantiatedExpression) => A)(name: String, expression: WomExpression, inputMapping: Map[String, OutputPort]): ErrorOr[A] = {
```

Comment:
Kind of surprising `A` can't be constrained any more than as a subtype of `GraphNode`.  It seems like there's a missing type there, even if it's just a marker trait.